Sources/KituraNet/HTTP/HTTPServerRequest.swift

```diff
@@ -41,18 +41,24 @@ public class HTTPServerRequest: ServerRequest {
 
     @available(*, deprecated, message: "URLComponents has a memory leak on linux as of swift 3.0.1. use 'urlURL' instead")
     public var urlComponents: URLComponents {
-        return URLComponents(url: urlURL, resolvingAgainstBaseURL: false) ?? URLComponents()
+        if let _urlComponents = _urlComponents {
+            return _urlComponents
+        } else {
+            return URLComponents(url: urlURL, resolvingAgainstBaseURL: false) ?? URLComponents()
+        }
     }
 
     private var _url: URL?
 
+    private var _urlComponents: URLComponents?
+
     public var urlURL: URL {
         if let _url = _url {
             return _url
         }
-        var url = ""
 
-        self.enableSSL ? url.append("https://") : url.append("http://")
+        _urlComponents = URLComponents()
```

Comment:
check if `_urlComponents` already exists here, we could have initialised it in `urlComponents`